src/rard/research/views/work.py

```diff
@@ -183,6 +217,39 @@ def get_context_data(self, *args, **kwargs):
         return context
 
 
+class BookUpdateIntroductionView(BookUpdateView):
+    model = Book
+    form_class = BookIntroductionForm
+    permission_required = ("research.change_book",)
+    template_name = "research/book_form.html"
+
+    def create_intro_if_does_not_exist(self, *args, **kwargs):
+        book = self.get_object()
+        if book.introduction is None:
+            book.introduction = TextObjectField.objects.create(content="")
+            book.save()
+
+    def dispatch(self, request, *args, **kwargs):
+        self.create_intro_if_does_not_exist()
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_success_url(self, *args, **kwargs):
+        return reverse("work:detail", kwargs={"pk": self.get_object().work.pk})
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context.update(
+            {
+                "editing": "introduction",
+                "change_work": True,
+                "change_book": True,
+                "has_object_lock": True,
+                "instance": self.get_object(),
```

Comment:
What are these for? The only one I can see being used in the template is has_object_lock and usually that's worked out in the template as
`{% with request.user|has_lock:object as has_object_lock %}`

Reply:
since the book update redirects to a new page (there's no book detail page) it loses some of the context (like editing permissions)
it's possible that this is no longer the case - I can try removing parts and seeing if it can be slimmed down